src/Microsoft.AspNetCore.Blazor.Browser.JS/src/Services/UriHelper.ts

```diff
@@ -1,9 +1,8 @@
-import { platform } from '../Environment';
-import { MethodHandle, System_String } from '../Platform/Platform';
-const registeredFunctionPrefix = 'Microsoft.AspNetCore.Blazor.Browser.Services.BrowserUriHelper';
-let notifyLocationChangedMethod: MethodHandle;
 let hasRegisteredEventListeners = false;
 
+// Will be initialized once someone registers
+let notifyLocationChangedCallback: { assemblyName: string, functionName: string } | null;
```

Comment:
Probably won't make any difference, but there's some potential for developer confusion here, which is really the TypeScript compiler's fault. The variable is typed as `somestuff | null` but its actual value will be `undefined` until initalized (which is neither `somestuff` nor `null`). I don't think the compiler should be allowing this.

Confusion would occur if someone writes elsewhere:

```
if (notifyLocationChangedCallback !== null) {
    // Do something assuming that notifyLocationChangedCallback is a "somestuff"
    // but it could fail because notifyLocationChangedCallback===undefined
}
```

Suggested fix:

```
let notifyLocationChangedCallback: { assemblyName: string, functionName: string } | null = null;
```

Reply:
Sure